### PR TITLE
#11340: migrate move to ttnn

### PR DESCRIPTION
**Pull request by sjameelTT**
### Ticket
#11340 

### Problem description
Move must be moved

### What's changed
copy paste

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: ttnn/cpp/ttnn/operations/data_movement/move/move_pybind.hpp [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <string>
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/data_movement/move/move_pybind.hpp:7]: Remove

File: ttnn/cpp/ttnn/operations/data_movement/move/move.hpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+#pragma once
+
+#include "ttnn/decorators.hpp"
+#include <optional>
+
+namespace ttnn {
+namespace operations::data_movement {
+
+static Tensor move_sharded(uint8_t queue_id, const Tensor& input_tensor, const std::optional<MemoryConfig>& mem_config);
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/data_movement/move/move.hpp:13]: Can these two be removed from header?

File: ttnn/cpp/ttnn/operations/data_movement/move/move.hpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        const Tensor& input_tensor,
+        const std::optional<MemoryConfig>& output_mem_config = std::nullopt);
+
+
+    static ttnn::Tensor invoke(
+        const Tensor& input_tensor,
+        const std::optional<MemoryConfig>& output_mem_config = std::nullopt);
+};
+
+namespace detail {
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/data_movement/move/move.hpp:28]: Can this be removed from header?

**Review by ayerofieiev-tt — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/data_movement/move/move_pybind.hpp (hunk 1 above)
Comment (sjameelTT) [reply ttnn/cpp/ttnn/operations/data_movement/move/move_pybind.hpp:7]: Done

File: ttnn/cpp/ttnn/operations/data_movement/move/move.hpp (hunk 2 above)
Comment (sjameelTT) [reply ttnn/cpp/ttnn/operations/data_movement/move/move.hpp:13]: Done

File: ttnn/cpp/ttnn/operations/data_movement/move/move.hpp (hunk 3 above)
Comment (sjameelTT) [reply ttnn/cpp/ttnn/operations/data_movement/move/move.hpp:28]: Dibe

**Review by arakhmati — APPROVED** [review]

**Comment by arakhmati** [comment]
"Move must be moved" 💀

**Review by esmalTT — APPROVED** [review]

**Review by tt-aho — APPROVED** [review]
bert + resnet look okay

**Review by djordje-tt — APPROVED** [review]

**Review by ntarafdar — APPROVED** [review]

**Pull request merged**